src/lib.rs

```diff
@@ -3133,6 +3133,52 @@ pub trait Itertools: Iterator {
         self.k_largest_by(k, k_smallest::key_to_cmp(key))
     }
 
+    /// Consumes the iterator and return an iterator of the last `n` elements.
+    ///
+    /// It allocates up to `n` elements.
+    /// The iterator, if directly collected to a `Vec`, is converted
+    /// without any extra copying or allocation cost.
+    ///
+    /// ```
+    /// use itertools::{assert_equal, Itertools};
+    ///
+    /// let v = vec![5, 9, 8, 4, 2, 12, 0];
+    /// assert_equal(v.iter().tail(3), &[2, 12, 0]);
+    /// assert_equal(v.iter().tail(10), &v);
+    ///
+    /// assert_equal((0..100).tail(10), 90..100);
+    /// ```
+    ///
+    /// For double ended iterators without side-effects, you might prefer
+    /// `.rev().take(n).collect_vec().into_iter().rev()`
+    /// to have the same result without consuming the entire iterator.
+    #[cfg(feature = "use_alloc")]
+    fn tail(self, n: usize) -> VecIntoIter<Self::Item>
+    where
+        Self: Sized,
+    {
+        match n {
+            0 => {
+                self.last();
+                Vec::new()
+            }
+            1 => self.last().into_iter().collect(),
+            _ => {
+                let mut iter = self.fuse();
+                let mut data: Vec<_> = iter.by_ref().take(n).collect();
+                // Update `data` cyclically.
+                let idx = iter.fold(0, |i, val| {
+                    data[i] = val;
+                    (i + 1) % n
+                });
+                // Respect the insertion order.
+                data.rotate_left(idx);
```

Comment:
This rotate the slice only once for the entire iteration.
But unless I don't know something, `VecDeque` would rotate the data multiple times during a long iteration to make some place on the back by pushing data on the front.

Reply:
The size of the allocation of a `VecDeque` necessarily is more than `n` to have some place to amortize the rotation (and it's not contiguous).
While my solution can allocate exactly `n` (in some cases) and the rotation is a simple clean-up at the end.

Reply:
Our MSRV is 1.43.1 (old). Would it make sense to keep my `Vec` implementation and switch to `VecDeque` when the MSRV reaches 1.67 ?

Reply:
I took a look at the new implementation, nice!

If we were to change `VecIntoIter` to `VecDequeIntoIter` once our MSRV is big enough, I suppose it would be a breaking change?!

Reply:
Or directly use `VecDeque`, I guess it's simpler.